examples/ex25p.cpp

```diff
@@ -0,0 +1,329 @@
+   {
+      for (int i = 0; i < bfs.Size(); ++i)
+      {
+         delete bfs[i];
+      }
+
+      for (int i = 0; i < essentialTrueDofs.Size(); ++i)
+      {
+         delete essentialTrueDofs[i];
+      }
+
+      delete fespace_lor;
+      delete fec_lor;
+      delete pmesh_lor;
+   }
+
+   void EliminateBCs(Vector& x, Vector& b, Vector& X, Vector& B)
+   {
+      OperatorPtr oper;
+      bfs.Last()->FormLinearSystem(*essentialTrueDofs.Last(), x, b,
+                                   oper, X, B);
+   }
+
+   void RecoverFEMSolution(const Vector& X, const Vector& b, Vector& x)
+   {
+      bfs.Last()->RecoverFEMSolution(X, b, x);
+   }
+
+private:
+   void AddIntegrators(BilinearForm* form)
+   {
+      form->AddDomainIntegrator(new DiffusionIntegrator(one));
+   }
+};
+
+int main(int argc, char *argv[])
+{
+   // 1. Initialize MPI.
+   int num_procs, myid;
+   MPI_Init(&argc, &argv);
+   MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
+   MPI_Comm_rank(MPI_COMM_WORLD, &myid);
+
+   // 2. Parse command-line options.
+   const char *mesh_file = "../data/star.mesh";
+   int geometricrefinements = 0;
+   int orderrefinements = 2;
+   const char *device_config = "cpu";
+   bool visualization = true;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&geometricrefinements, "-gr", "--geometricrefinements",
+                  "Number of geometric refinements done prior to order refinements.");
+   args.AddOption(&orderrefinements, "-or", "--orderrefinements",
+                  "Number of order refinements. Finest level in the hierarchy has order 2^{or}.");
+   args.AddOption(&device_config, "-d", "--device",
+                  "Device configuration string, see Device::Configure().");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (myid == 0)
+      {
+         args.PrintUsage(cout);
+      }
+      MPI_Finalize();
+      return 1;
+   }
+   if (myid == 0)
+   {
+      args.PrintOptions(cout);
+   }
+
+   // 3. Enable hardware devices such as GPUs, and programming models such as
+   //    CUDA, OCCA, RAJA and OpenMP based on command line options.
+   Device device(device_config);
+   if (myid == 0) { device.Print(); }
+
+   // 4. Read the (serial) mesh from the given mesh file on all processors.  We
+   //    can handle triangular, quadrilateral, tetrahedral, hexahedral, surface
+   //    and volume meshes with the same code.
+   Mesh *mesh = new Mesh(mesh_file, 1, 1);
+   int dim = mesh->Dimension();
+
+   // 5. Refine the serial mesh on all processors to increase the resolution. In
+   //    this example we do 'ref_levels' of uniform refinement. We choose
+   //    'ref_levels' to be the largest number that gives a final mesh with no
+   //    more than 10,000 elements.
```

Comment:
```suggestion
   //    more than 1,000 elements.
```

(there is a typo either in the comment or in the code — probably 1,000 is better for the serial refinements, since you do two parallel refinements later)